gcc/rust/resolve/rust-ast-resolve-expr.cc

```diff
@@ -120,6 +120,11 @@ ResolveExpr::visit (AST::IdentifierExpr &expr)
     {
       resolver->insert_resolved_type (expr.get_node_id (), resolved_node);
     }
+  else if (funny_error)
+    {
+      rust_error_at (expr.get_locus (), "oopsie woopsie! you broke GCC Rust");
+      internal_error_no_backtrace ("you broke me");
```

Comment:
I think we could use a `rust_fatal_error` here instead and not ICE - the ICE function prompts the user to submit a full bug report, which we don't want here
```suggestion
      rust_fatal_error (expr.get_locus (),
			"are you trying to break %s? how dare you?",
			expr.as_string ().c_str ());
```
outputs with the following message:

```
arthur@platypus ~/G/r/gccrs ((8657790c))> build/gcc/crab1 main.rs
main.rs:2:11: fatal error: are you trying to break rust? how dare you?
    2 |     break rust;
      |           ^~~~
compilation terminated.
```
which I think is less misleading than the current output

```
arthur@platypus ~/G/r/gccrs ((8657790c)) [1]> build/gcc/crab1 main.rs
main.rs:2:11: error: oopsie woopsie! you broke GCC Rust
    2 |     break rust;
      |           ^~~~
crab1: internal compiler error: you broke me
Please submit a full bug report, with preprocessed source (by using -freport-bug).
See <https://gcc.gnu.org/bugs/> for instructions.
```

Reply:
> I think we could use a `rust_fatal_error` here instead and not ICE

No, we really want this to be an ICE. The backstory here is that rustc [used to ICE](https://github.com/rust-lang/rust/issues/43162) (not just emit a regular error) when type-checking `break some_expr()` outside of a loop. This was fixed (to just emit an error), but `break rust` was intentionally kept an ICE, as a fun Easter egg. So we need to ICE too, a regular error won't cut it.

> the ICE function prompts the user to submit a full bug report, which we don't want here

Yes. But maybe we can get the best of both worlds:

```
$ gcc -frust-incomplete-and-experimental-compiler-do-not-use /tmp/break-rust3.rs
/tmp/break-rust3.rs:2:11: internal compiler error: are you trying to break rust? how dare you?
    2 |     break rust;
      |           ^~~~
You broke GCC Rust. This is a feature.
```
Yes, this is an ICE with rich location. How cool is that!

(Also, "broke" or "have broken"?)

Reply:
> (Also, "broke" or "have broken"?)

Went for "have broken", but this can be reverted if "broke" is better / more correct.